dart/lib/flat_buffers.dart

```diff
@@ -887,8 +915,15 @@ class Float64ListReader extends Reader<List<double>> {
 
   @override
   @pragma('vm:prefer-inline')
-  List<double> read(BufferContext bc, int offset) =>
-      _FbFloat64List(bc, bc.derefObject(offset));
+  List<double> read(BufferContext bc, int offset) {
+    if (Endian.host == Endian.little) {
+        final listOffset = bc.derefObject(offset);
+        final length = bc._getUint32(listOffset);
+        return bc._asFloat64List(listOffset + _sizeofUint32, length);
+	  } else {
+        return _FbFloat64List(bc, bc.derefObject(offset));
+	  }
+  }
```

Comment:
With this change, test_writeList_ofFloat64 is failing as the offset must be a multiple of BYTES_PER_ELEMENT (8). I'm absolutely missing something simple here, need to wrap my head around what it is (or grab another coffee, maybe).